sklearn/cluster/k_means_.py

```diff
@@ -338,12 +338,13 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
     if n_jobs == 1:
         # For a single thread, less memory is needed if we just store one set
         # of the best results (as opposed to one set per run per thread).
-        for it in range(n_init):
+        seeds = random_state.randint(np.iinfo(np.int32).max, size=n_init)
```

Comment:
You can move the `seeds` assignment outside of the `if` clause since it used both for `n_job == 1` and `n_jobs != 1`.